### PR TITLE
Use contextual tuples for admin role

### PR DESCRIPTION
**Pull request by nsklikas**
IAM-939

I have tried using the UI with this and it works, but I am not sure how to actually test that these changes are working properly.

### CODE REVIEW

**Review by BarcoMasile — DISMISSED** [review]

**Review by BarcoMasile — APPROVED** [review]

**Review by shipperizer — COMMENTED** [review]
the change looks good, i m just wondering if the aim of the PR was to only allow contextual tuples on the roles endpoints or everywhere?

shouldnt we add this functionality on every single converter?

**Review by BarcoMasile — DISMISSED** [review]

**Review by BarcoMasile — APPROVED** [review]

File: internal/authorization/converters_test.go [hunk 1]
```diff
@@ -480,49 +582,91 @@ func TestRoleConverterMapReturnsPermissions(t *testing.T) {
...
+				},
 			},
 		},
 		{
 			name:  "DELETE /api/v0/roles/id-1234/entitlements/can_view::role:1",
 			input: input{method: http.MethodDelete, endpoint: "/api/v0/roles/id-1234/entitlements", ID: "id-1234", EntitlementID: "can_view::role:1"},
 			output: []Permission{
-				{Relation: CAN_EDIT, ResourceID: fmt.Sprintf("%s:%s", ROLE_TYPE, "id-1234")},
+				{
+					Relation:   CAN_EDIT,
```
Comment (natalian98) [inline internal/authorization/converters_test.go:638]:
```suggestion
					Relation:   CAN_DELETE,
```

File: internal/authorization/converters_test.go [hunk 2]
```diff
@@ -595,80 +739,146 @@ func TestGroupConverterMapReturnsPermissions(t *testing.T) {
...
+				},
 			},
 		},
 		{
 			name:  "DELETE /api/v0/groups/id-1234/entitlements/can_view::role:1",
 			input: input{method: http.MethodDelete, endpoint: "/api/v0/groups/id-1234/entitlements", ID: "id-1234", EntitlementID: "can_view::role:1"},
 			output: []Permission{
-				{Relation: CAN_EDIT, ResourceID: fmt.Sprintf("%s:%s", GROUP_TYPE, "id-1234")},
+				{
+					Relation:   CAN_EDIT,
```
Comment (natalian98) [inline internal/authorization/converters_test.go:795]:
```suggestion
					Relation:   CAN_DELETE,
```

File: internal/authorization/converters_test.go [hunk 3]
```diff
@@ -595,80 +739,146 @@ func TestGroupConverterMapReturnsPermissions(t *testing.T) {
...
+				},
 			},
 		},
 		{
 			name:  "DELETE /api/v0/groups/id-1234/roles/viewer",
 			input: input{method: http.MethodDelete, endpoint: "/api/v0/groups/id-1234/roles", ID: "id-1234", RoleID: "viewer"},
 			output: []Permission{
-				{Relation: CAN_EDIT, ResourceID: fmt.Sprintf("%s:%s", GROUP_TYPE, "id-1234")},
+				{
+					Relation:   CAN_EDIT,
```
Comment (natalian98) [inline internal/authorization/converters_test.go:848]:
```suggestion
					Relation:   CAN_DELETE,
```

**Review by natalian98 — APPROVED** [review]

**Pull request merged**